packaging/ubuntu-14.04/rules

```diff
@@ -35,14 +35,19 @@ SYSTEMD_UNITS_DESTDIR="lib/systemd/upstart/"
 # work around that by constructing the appropriate -I flag by hand.
 GCCGO := $(shell go tool dist env > /dev/null 2>&1 && echo no || echo yes)
 
+BUILDFLAGS:=-pkgdir=$(CURDIR)/_build/std
 # Disable -buildmode=pie mode on i386 as can panics in spectacular
 # ways (LP: #1711052).
 # See also https://forum.snapcraft.io/t/artful-i386-panics/
 # Note while the panic is only on artful, that's because artful
 # detects it; the issue potentially there on older things.
-BUILDFLAGS:=-pkgdir=$(CURDIR)/_build/std
+#
+# Disable -buildmode=pie mode on armhf as this may lead to nasty
+# memory exaustion issues (LP: #1822738)
 ifneq ($(shell dpkg-architecture -qDEB_HOST_ARCH),i386)
```

Comment:
Perhaps something like this:

```
DEB_HOST_ARCH := $(shell dpkg-architecture -qDEB_HOST_ARCH)
ifneq(,$(or $(findstring i386,$(DEB_HOST_ARCH)),$(findstring armhf,$(DEB_HOST_ARCH))))
BUILDFLAGS+= -buildmode=pie
endif
```

Reply:
I will rework this to use "dpkg-architecture -qDEB_TARGET_ARCH_BITS" - however this logic here seems slightly off:
```
$ cat /tmp/lala.mk && printf "\n\nresult:\n" && make -f  /tmp/lala.mk
DEB_HOST_ARCH := i386
ifneq (,$(or $(findstring i386,$(DEB_HOST_ARCH)),$(findstring armhf,$(DEB_HOST_ARCH))))
BUILDFLAGS+= -buildmode=pie
endif

all:
	echo ${BUILDFLAGS}


result:
echo -buildmode=pie
-buildmode=pie
```
Anyway, not important as the dpkg-architecture call will simplify this (why is make so hard for this kind of tests :(
```